tests/shared/src/test/scala/scala/meta/tests/DottyEnumTests.scala

```diff
@@ -0,0 +1,192 @@
+      """List(Init(Type.Name("Color"), Name(""), List(List(Lit.Int(0xFF0000)))))), Defn.Enum.Case(Nil, Term.Name("Green"), Nil, """+
+      """Ctor.Primary(Nil, Name(""), Nil), List(Init(Type.Name("Color"), Name(""), List(List(Lit.Int(0x00FF00)))))), """+
+      """Defn.Enum.Case(Nil, Term.Name("Blue"), Nil, Ctor.Primary(Nil, Name(""), Nil), List(Init(Type.Name("Color"), Name(""), """+
+      """List(List(Lit.Int(0x0000FF)))))))))))"""
+    )
+
+    checkBad("enum {}")
+
+    checkBad("enum ; Foo{}")
+
+    checkBad(
+      """
+      |enum Foo{
+      | case A(x : Int), B
+      |}
+      """.stripMargin
+    )
+
+    checkBad(
+      """
+      |enum Foo{
+      | case A, B(x : Int)
+      |}
+      """.stripMargin
+    )
+
+    checkBad(
+      """
+      |enum Foo{
+      | case A,B(x:Int)
+      |}
+      """.stripMargin
+    )
+
+    checkBad("enum Foo Bar{}")
+
+    checkOKParse(
+      """
+      |enum Foo{
+      | case A(x : Int)
+      | class A{
+      |   def foo : Int = 2
+      | }
+      |}
+      """.stripMargin
+    )
+
+    checkOK(
+      """
+      |enum Foo{
+      | def foo : Int = 2
+      | case A
+      |}
+      """.stripMargin,
+      """Source(List(Defn.Enum(Nil, Type.Name("Foo"), Nil, Ctor.Primary(Nil, Name(""), Nil), """+
+      """Template(Nil, Nil, Self(Name(""), None), List(Defn.Def(Nil, Term.Name("foo"), """+
+      """Nil, Nil, Some(Type.Name("Int")), Lit.Int(2)), Defn.Enum.Case(Nil, Term.Name("A"), Nil, """+
+      """Ctor.Primary(Nil, Name(""), Nil), Nil))))))"""
+    )
+
+    checkOKParse(
+      """
+      |enum Planet(mass: Double, radius: Double) {
+      |  private final val G = 6.67300E-11
+      |  def surfaceGravity = G * mass / (radius * radius)
+      |  def surfaceWeight(otherMass: Double) =  otherMass * surfaceGravity
+      |  case MERCURY extends Planet(3.303e+23, 2.4397e6)
+      |  case VENUS   extends Planet(4.869e+24, 6.0518e6)
+      |  case EARTH   extends Planet(5.976e+24, 6.37814e6)
+      |  case MARS    extends Planet(6.421e+23, 3.3972e6)
+      |  case JUPITER extends Planet(1.9e+27,   7.1492e7)
+      |  case SATURN  extends Planet(5.688e+26, 6.0268e7)
+      |  case URANUS  extends Planet(8.686e+25, 2.5559e7)
+      |  case NEPTUNE extends Planet(1.024e+26, 2.4746e7)
+      |}
+      """.stripMargin
+    )
+
+    checkOK(
+      """
+      |enum Foo{
+      | case A
+      | class A{}
+      |}
+      """.stripMargin,
+      """Source(List(Defn.Enum(Nil, Type.Name("Foo"), Nil, Ctor.Primary(Nil, Name(""), Nil), """+
+      """Template(Nil, Nil, Self(Name(""), None), List(Defn.Enum.Case(Nil, Term.Name("A"), Nil, """+
+      """Ctor.Primary(Nil, Name(""), Nil), Nil), Defn.Class(Nil, Type.Name("A"), Nil, """+
+      """Ctor.Primary(Nil, Name(""), Nil), Template(Nil, Nil, Self(Name(""), None), Nil)))))))"""
+    )
+
+    checkOKParse(
+      """
+      |enum Foo{
+      | case A
+      | enum Bar{
+      |   case B  
+      |}
+      |}
+      """.stripMargin
+    )   
+}
```

Comment:
I think we need more test case. Some ideas:
- not ok: `enum Foo { case Bar { def hello = 1 } }`
- ok:
  ```scala
  enum Foo {
    case X,
         Y,
         Z
  }
  ```
- ok: `enum Bar extends X with Y with Z {}`

Try to think of more invalid test cases. For example where it could interact badly with existing feature (e.g. cases in pattern match)